### PR TITLE
create structure, some endpoints and models, def token system

### PR DESCRIPTION
**Pull request by keyzoz**
i will change token system and add aws in the future

### CODE REVIEW

File: setup.cfg
Comment (sabbra-cadabra) [inline setup.cfg:1]: why do you need setup.cfg?

File: docker-compose-local.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+version: "3.0"
+services:
+  db:
+    container_name: "db"
+    image: postgres:15
+    restart: always
+    environment:
+      - POSTGRES_DB=postgres
```
Comment (sabbra-cadabra) [inline docker-compose-local.yaml:8]: you may want to use .env file for your variables

File: docker-compose-local.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+version: "3.0"
+services:
+  db:
+    container_name: "db"
```
Comment (sabbra-cadabra) [inline docker-compose-local.yaml:4]: if you explicitly give a name for your container it makes sense to pass another name. it you want it to be called the same this option is redundant

File: docker-compose-local.yaml
Comment (sabbra-cadabra) [inline docker-compose-local.yaml:1]: I can't find FastAPI app in this file. Please add to run all your services together

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export DOCKER_DEFAULT_PLATFORM=linux/arm64/v8
```
Comment (sabbra-cadabra) [inline Makefile:1]: do you have any requirements for a specific platform?

File: requirements.txt
Comment (sabbra-cadabra) [inline requirements.txt:1]: Let's use pipenv or poetry for managing dependencies.

File: settings.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
+from envparse import Env
+
+env = Env()
+
+REAL_DATABASE_URL = env.str(
+    "REAL_DATABASE_URL",
+    default="postgresql+asyncpg://postgres:postgres@0.0.0.0:5444/postgres",
```
Comment (sabbra-cadabra) [inline settings.py:7]: please do not hardcode any database settings! you must use .env.

File: src/api/actions/user.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
+from uuid import UUID
+
+from src.api.schemas import ShowUser, UserCreate
+from src.db.dals import Roles, UserDAL
+from src.db.models import User
+from src.hashing import Hasher
+
+
+async def _create_new_user(body: UserCreate, session) -> ShowUser:
```
Comment (sabbra-cadabra) [inline src/api/actions/user.py:9]: I have a strong feeling you must employ Class based CRUD. Also think of Ports and Adapters pattern from the requirements.

File: src/api/routers/user_router.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
+from logging import getLogger
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException
+from fastapi_jwt_auth import AuthJWT
+from sqlalchemy.exc import IntegrityError
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.api.actions.auth import get_current_user_from_token
+from src.api.actions.user import (_create_new_user, _delete_user,
```
Comment (sabbra-cadabra) [inline src/api/routers/user_router.py:10]: instead of using them as single functions please use a UserCRUD class.

**Review by sabbra-cadabra — CHANGES_REQUESTED** [review]
Something seems off about your project structure. Did you do it by yourself or used another repository as an example?

File: setup.cfg
Comment (keyzoz) [reply setup.cfg:1]: if i will need some setups for my pre-commits

File: src/api/routers/user_router.py [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        raise HTTPException(status_code=503, detail=f"Database error: {err}")
+
+
+@user_router.patch("/me")
+async def update_current_user(
+    body: UpdateUser,
+    Authorize: AuthJWT = Depends(),
+    session: AsyncSession = Depends(get_db),
+) -> ShowUser:
+    try:
```
Comment (sabbra-cadabra) [inline src/api/routers/user_router.py:36]: you can put full block in one try ... except

**Review by sabbra-cadabra — APPROVED** [review]

**Pull request closed without merging**